### PR TITLE
Irs expats

### PR DESCRIPTION
**Pull request by kghamilton89**

### CODE REVIEW

**Comment by rodionos** [comment]
The aggregate data for the last quarter was missing. I mean it was inserted but it was inserted into a different metric (not us-expatriate-aggregate which you used for charts). I re-inserted it, and now screenshots and chartlab examples have to be redone. I think people and percentages need to be on different axis. We should also limit the displayed area for percentage change to -100% to +100%. I'll revisit the article once the analysis is updated.

File: Expatriation/2018-1.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+| 2008 | 229        | -241.00      | -51.28          |
+| 2009 | 741        | 512.00       | 223.58          |
+| 2010 | 1531       | 790.00       | 106.61          |
+| 2011 | 1780       | 249.00       | 16.26           |
+| 2012 | 932        | -848.00      | -47.64          |
+| 2013 | 2999       | 2067.00      | 221.78          |
+| 2014 | 3411       | 412.00       | 13.74           |
+| 2015 | 4273       | 862.00       | 25.27           |
+| 2016 | 5398       | 1125.00      | 26.33           |
+| 2017 | 5128       | -270.00      | -5.00           |
```
Comment (rodionos) [inline Expatriation/2018-1.md]: To check these numbers, you can run this query:

```sql
SELECT datetime, count(*)
FROM "us-expatriate-counter"
  WHERE entity = 'us.irs'
GROUP BY datetime
```

The last 4 quarters are:

```
2017-06-30T00:00:00Z | 1756
2017-09-30T00:00:00Z | 1374
2017-12-31T00:00:00Z | 685
2018-03-31T00:00:00Z | 1098
```

Therefore the query above should display `1756+1374+685+1098` = `4913` with the date being 2018 and meaning that this is the total for 4 quarters of each year ending with the current quarter. Hint: current quarter ends on **March 31, 2018**. 

> It's a always a good practice to check your numbers by hand from the ground up.

File: Expatriation/2018-1.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+| 3       | 2015 | 1426          |
+| 4       | 2015 | 1057          |
+| 1       | 2016 | 1154          |
+| 2       | 2016 | 506           |
+| 3       | 2016 | 1379          |
+| 4       | 2016 | 2359          |
+| 1       | 2017 | 1313          |
+| 2       | 2017 | 1756          |
+| 3       | 2017 | 1374          |
+| 4       | 2017 | 685           |
```
Comment (rodionos) [inline Expatriation/2018-1.md]: The query is not showing `1098` for the 1st quarter of 2018.

File: Expatriation/2018-1.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
+# Tracking Expatriation with ChartLab, SQL, and Web Crawler in ATSD
+
+![](Images/expat-title.png)
+
+## Introduction
+
+Each year the Internal Revenue Service of the United States releases expatriation information about American citizens who have decided to renounce their citizenship via the [Federal Register](https://www.federalregister.gov/), the government's official journal. The reason the IRS publishes this information is to levy the [Expatriation Tax](https://www.irs.gov/individuals/international-taxpayers/expatriation-tax) upon those citizens with enough wealth to incur the tax at the time of their renunciation. The tax doesn't apply to just anyone who decides to leave the country for good, only to those soon-to-be-former citizens who qualify as "covered expatriates".
```
Comment (rodionos) [inline Expatriation/2018-1.md]: As you have mentioned, some of the content repeats from previous articles. Why not directly link to them and only talk about what's new this quarter? It's fine to have a brief article to deliver updated data and insights for this specific quarter.

Comment (kghamilton89) [reply Expatriation/2018-1.md]: Removed introduction and linked to previous articles.

File: Expatriation/2018-1.md (hunk 1 above)
Comment (kghamilton89) [reply Expatriation/2018-1.md]: Updated `enddate` parameter in query.

File: Expatriation/2018-1.md (hunk 2 above)
Comment (kghamilton89) [reply Expatriation/2018-1.md]: Updated title of this article and query to include 2018-Q1 data.

File: Expatriation/README.md [hunk 4]
```diff
@@ -1,125 +1,91 @@
...
 
-![](Images/expat-title.png)
+![](Images/title1-2018.png)
 
 ## Introduction
 
-Each year the Internal Revenue Service of the United States releases expatriation information about American citizens who have decided to renounce their citizenship via the [Federal Register](https://www.federalregister.gov/), the government's official journal. The reason the IRS publishes this information is to levy the [Expatriation Tax](https://www.irs.gov/individuals/international-taxpayers/expatriation-tax) upon those citizens with enough wealth to incur the tax at the time of their renunciation. The tax doesn't apply to just anyone who decides to leave the country for good, only to those soon-to-be-former citizens who qualify as "covered expatriates".
+The [Internal Revenue Service](https://www.irs.gov/) (IRS) of the United States releases quarterly reports tracking the number of expatriated citizens processed in the previous four-month period. They publish these reports via the [Federal Register](https://www.federalregister.gov/) which is the primary publication medium of the federal government and used by a number of agencies as an outreach arm to the public.
```
[the remark this answers is not in the document]
Comment (kghamilton89) [reply Expatriation/README.md]: My mistake.

File: Expatriation/README.md [hunk 5]
```diff
@@ -1,125 +1,91 @@
...
+---
 
-Both of the above graphs were prepared with Axibase ChartLab which is a visualization sandbox that features convenient syntax with different types of charts. The above visualizations may be modified to produce several projections of the same dataset stored in Axibase Time Series Database (ATSD).
+## Current Data
 
-A number of built-in [statistical functions](https://github.com/axibase/atsd/blob/master/rule-engine/functions-statistical.md) are already supported by ChartLab, and [user-defined functions](../how-to/shared/trends.md#user-defined-functions) may be added to a local ATSD instance. You can follow these [instructions](https://github.com/axibase/charts/blob/master/syntax/udf.md#deploying-function-files) to upload a user-defined function as a JavaScript file.
-
-![](Images/previous-year.png)
-[![](Images/btn.png)](https://apps.axibase.com/chartlab/a14a69a4#fullscreen)
+Annual and quarterly data may be queried to return samples stored in [Axibase Time Series Database](https://axibase.com/products/axibase-time-series-database/) using an SQL emulator in the user interface.
```
[the remark this answers is not in the document]
Comment (kghamilton89) [reply Expatriation/README.md]: Replaced.

File: Expatriation/README.md [hunk 6]
```diff
@@ -1,125 +1,91 @@
...
+| 2011 | 1850       | 997          | 117             |
+| 2012 | 1742       | -108         | -6              |
+| 2013 | 1151       | -591         | -34             |
+| 2014 | 3319       | 2168         | 188             |
+| 2015 | 3743       | 424          | 13              |
+| 2016 | 4096       | 353          | 9               |
+| 2017 | 5557       | 1461         | 36              |
+| 2018 | 4913       | -644         | -12             |
+
+Year-on-Year absolute and percentage changes have been omitted from current year data since the collected dataset remains incomplete, however, it's plain to see that 2018 expatriation numbers are on-track to potentially surpass the record-setting numbers of 2017.
```
[the remark this answers is not in the document]
Comment (kghamilton89) [reply Expatriation/README.md]: I just removed this whole paragraph.

File: Expatriation/README.md [hunk 7]
```diff
@@ -159,14 +132,79 @@ The Web Crawler reads incoming data from the Federal Register and parses it into
...
+
+```ls
+value = fred.PercentChangeFromYearAgo('raw')
+```
+
+> View the complete [`fred.js`](../how-to/shared/trends.md#fred-library) UDF library here.
+
+#### Alert Expression
+
+Customized data monitoring in ATSD is possible using [`alert-expressions`](https://axibase.com/products/axibase-time-series-database/visualization/widgets/time-chart/#tab-id-14) whereby user-specified parameters may be defined to trigger alarms based on incoming data.
```
[the remark this answers is not in the document]
Comment (kghamilton89) [reply Expatriation/README.md]: Removed.

File: Expatriation/README.md [hunk 8]
```diff
@@ -1,125 +1,91 @@
...
+| 2018 | 4913       | -644         | -12             |
+
+Year-on-Year absolute and percentage changes have been omitted from current year data since the collected dataset remains incomplete, however, it's plain to see that 2018 expatriation numbers are on-track to potentially surpass the record-setting numbers of 2017.
+
+[**ChartLab**](../ChartLabIntro/README.md) is a visualization service which is supported by data processing and storage tasks in ATSD. **ChartLab** features a wide-range of output features without robust syntax. Used here, expatriation data may be visualized to support the above SQL query.
+
+![](Images/2018-q2.png)
+
+[![](Images/btn.png)](https://apps.axibase.com/chartlab/7c5c7780#fullscreen)
+
```
[the remark this answers is not in the document]
Comment (kghamilton89) [reply Expatriation/README.md]: `endtime` expression modified in ChartLab visualization, screenshot updated.

File: Expatriation/README.md [hunk 9]
```diff
@@ -159,14 +132,79 @@ The Web Crawler reads incoming data from the Federal Register and parses it into
...
+---
+
+### Additional Visualizations
+
+#### `time_offset`
+
+Data may also be compared using `time_offset` features whereby variable time-offsets may be applied to a dataset so that it may be compared to itself during a different time period, useful when working with time series data.
+
+![](Images/2018-q2-3.png)
+
```
[the remark this answers is not in the document]
Comment (kghamilton89) [reply Expatriation/README.md]: It does work it was just incorrectly used. It should be `style = fill-opacity:1.0` not `style = opacity:1.0`. Updated syntax, replaced screenshots.

**Comment by rodionos** [comment]
I edited README because it was faster than fixing. Please review the changes (my diff), so that we can avoid them perhaps going forward. Review the final draft.

**Comment by kghamilton89** [comment]
Ready to merge.

**Pull request merged**